### PR TITLE
Improve lcov coverage parsing

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- extract line coverage percent from lcov files
- use the new parsing logic when running Rust coverage
- update tests for lcov coverage parsing

## Testing
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_688588977888832283e0264c1eb19b1e

## Summary by Sourcery

Improve coverage parsing by extracting line coverage from lcov and Cobertura outputs via new helper functions, integrating them into the Rust and Python coverage scripts, and bolstering error handling and test coverage.

New Features:
- Add get_line_coverage_percent_from_lcov to extract line coverage from lcov.info files
- Introduce coverage_parsers module with get_line_coverage_percent_from_cobertura for Cobertura XML

Enhancements:
- Update run_rust and run_python scripts to use the new parsers for lcov and Cobertura formats
- Include lxml dependency and improve error handling with proper exit codes and two-decimal formatting

Tests:
- Add unit tests for lcov parser covering zero lines, missing tags, malformed data, and file errors
- Add unit tests for Cobertura parser covering per-line detail, root totals fallback, and zero coverage cases

### CODE REVIEW

File: .github/actions/generate-coverage/tests/test_scripts.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def test_run_rust_success(tmp_path: Path, shell_stubs: StubManager) -> None:
     out = tmp_path / "cov.lcov"
     gh = tmp_path / "gh.txt"
 
+    out.write_text("LF:200\nLH:163\n")
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/tests/test_scripts.py:64]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 2]
```diff
@@ -48,6 +48,20 @@ def extract_percent(output: str) -> str:
...
+    text = lcov_file.read_text(encoding="utf-8")
+
+    def total(tag: str) -> int:
+        return sum(map(int, re.findall(rf"^{tag}:(\d+)$", text, flags=re.MULTILINE)))
+
+    lines_found = total("LF")
+    lines_hit = total("LH")
+    if lines_found == 0:
+        return "0.00"
+    return f"{lines_hit / lines_found * 100:.2f}"
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py:62]: Has this now been resolved?

**Comment by leynos** [comment]
@sourcery-ai review

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 3]
```diff
@@ -44,10 +45,55 @@ def extract_percent(output: str) -> str:
...
+
+    try:
+        lines_found = total("LF")
+        lines_hit = total("LH")
+    except ValueError:
+        return "0.00"
+
+    if lines_found == 0:
+        return "0.00"
+    return f"{lines_hit / lines_found * 100:.2f}"
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 4]
```diff
@@ -44,10 +45,55 @@ def extract_percent(output: str) -> str:
     )
     if not match:
         typer.echo("Could not parse coverage percent", err=True)
-        raise typer.Exit(code=1)
+        raise typer.Exit(1)
     return match[1]
 
 
+def get_line_coverage_percent_from_lcov(lcov_file: Path) -> str:
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_rust.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_python.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env -S uv run --script
 # /// script
 # requires-python = ">=3.12"
-# dependencies = ["plumbum", "typer", "defusedxml"]
+# dependencies = ["plumbum", "typer", "lxml"]
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_python.py:4]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_rust.py [hunk 6]
```diff
@@ -48,6 +48,73 @@ def extract_percent(output: str) -> str:
...
+        total = num_or_zero("/coverage/@lines-valid")
+
+    if total == 0:
+        return "0.00"
+
+    percent = (
+        Decimal(covered) / Decimal(total) * 100
+    ).quantize(Decimal("0.01"), rounding=ROUND_HALF_UP)
+    return f"{percent}"
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

File: .github/actions/generate-coverage/scripts/run_rust.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Please supply a prompt for an AI coding agent to enable it to apply the fix.

File: .github/actions/generate-coverage/scripts/run_rust.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?

[the remark this answers is not in the document]
Comment (leynos) [reply .github/actions/generate-coverage/scripts/run_rust.py]: Has this now been resolved?
